addons/account/models/account_move.py

```diff
@@ -816,7 +816,12 @@ def _compute_highest_name(self):
 
     @api.model
     def _get_query_made_hole(self, ids=None):
-        return f"""
+        ids_domain = SQL()
+        if ids:
+            ids_domain = SQL("AND this.id = ANY(%s)", ids)
+        elif irregular_domain := self.env.context.get('irregular_sequence_domain'):
+            ids_domain = SQL("AND this.id IN (%s)", self._where_calc(irregular_domain).subselect())
```

Comment:
@oco-odoo fixed the import error, and also changed the `.select()` into `.subselect()` here